File: Lua/Pseudocode/5 Problems/EvenOdds.lua
--random number 10 times, see how many are even and how many are odd
function printEvenOdds()
    --variables
    local zeroCount = 0
    local evenCount = 0
    local oddCount = 0
    local checkNum = nil

    --Setup random number generator
    math.randomseed(os.time()) -- seed the randomizer
    math.random() ; math.random() ; math.random() -- get rid of any old values

    for i = 1, 10, 1 do
        checkNum = math.random(0, 100)

        if (checkNum == 0) then
            zeroCount = zeroCount + 1
            print(checkNum.." is zero")
        elseif (checkNum%2 == 0) then -- 1 means odd, 0 means even
            evenCount = evenCount + 1
            print(checkNum.." is even")
        else
            oddCount = oddCount + 1
            print(checkNum.." is odd")
        end
    end

    print("Zeroes: "..zeroCount)
    print("Left: "..oddCount)
    print("Right: "..evenCount)
end

File: Python/Pseudocode/5 Problems/EvenOdds.py
# Get a random number. If the number is even print “RIGHT”, if it’s odd, print “LEFT”. Do this ten times. Print out how many times it was “LEFT” and how many times it was “RIGHT”.
import random

# Initialize variables
zeroCount = 0
evenCount = 0
oddCount = 0

# Run 10 times
for i in range(0, 10):
  # Random number 0 to 100
  checkNum = random.randint(0, 100)
  if checkNum == 0: # Zero
    zeroCount += 1
    print(str(checkNum) + " is zero")
  elif checkNum % 2 == 0: # Even
    evenCount += 1
    print(str(checkNum) + " is even")
  else: # Odd
    oddCount += 1
    print(str(checkNum) + " is odd")

# Print results
print("Zeros: " + str(zeroCount))
print("Left: " + str(evenCount))
print("Right: " + str(oddCount))



